cpp/src/io/json/json_column.cu

```diff
@@ -541,9 +504,31 @@ void make_device_json_column(device_span<SymbolT const> input,
       }
     });
 
-  // restore col_ids, TODO is this required?
-  // thrust::copy(
-  //   rmm::exec_policy(stream), original_col_ids.begin(), original_col_ids.end(), col_ids.begin());
+  // 4. scatter string offsets to respective columns, set validity bits
+  thrust::for_each_n(
+    rmm::exec_policy(stream),
+    thrust::counting_iterator<size_type>(0),
+    col_ids.size(),
+    [node_categories = tree.node_categories.begin(),
+     col_ids         = original_col_ids.begin(),
+     row_offsets     = row_offsets.begin(),
+     range_begin     = tree.node_range_begin.begin(),
+     range_end       = tree.node_range_end.begin(),
+     d_ignore_vals   = d_ignore_vals.begin(),
+     d_columns_data  = d_columns_data.begin()] __device__(size_type i) {
+      switch (node_categories[i]) {
+        case NC_STRUCT: set_bit(d_columns_data[col_ids[i]].validity, row_offsets[i]); break;
+        case NC_LIST: set_bit(d_columns_data[col_ids[i]].validity, row_offsets[i]); break;
+        case NC_VAL:
```

Comment:
Please use an explicit `[[fallthrough]]` any time that a `case` statement has no body and uses the subsequent `case`.
```suggestion
        case NC_VAL: [[fallthrough]];
```